adr/minimal-sdk.md

```diff
@@ -0,0 +1,54 @@
+# ADR: Adopt a Minimal SDK and Ecosystem Strategy for Agents
+
+## Context
+
+Following the decision in [ADR: Adopt Llama Stack Responses API for Agentic MCP](https://github.com/opendatahub-io/agents/pull/1) to adopt the Llama Stack Responses API, we committed to providing a client-side "Agent construct" to manage state and orchestrate the agentic loop. This ADR addresses the critical follow-up question: what specific form will this construct take?
+
+The goal is to provide a solution that enables developers, offers a clear path for simple use cases, and integrates well with the broader AI ecosystem, all while managing Red Hat's engineering investment and long-term strategy.
+
+We considered several options for implementing this agent construct:
+
+1. **Extend Llama Stack Client SDK:** Add the agent logic directly into the official Llama Stack client. This would create a strong, unified story but conflicts with the client's design as a thin, auto-generated layer.
+2. **New, Full-Featured Red Hat SDK:** Build a comprehensive, competitive agent framework from the ground up. This offers maximum control but requires a very significant, long-term engineering investment. Also, even if it was done well, it would be difficult to drive adoption due to strong competition in this space.
+3. **New, Full-Featured Red Hat Container/Service:** The same as above, but delivered as a containerized HTTP service. This could simplify agent sharing and support a broader assortment of programming languages, but it adds operational complexity for developers.
```

Comment:
Comment: Also, if it belongs on the server side, we would strongly consider just adding it to the Llama Stack server.